### ISSUE TITLE
Add API and UI  support for constrained folding

### ISSUE THREAD

**Issue by AntonPetrov** [RNAcentral/R2DT#74]
Hi Fabio @biomadeira!

The latest R2DT version that you already deployed on wwwdev supports several new command line options that allow the users to perform additional folding for the regions which are left unfolded using the templates (see the [docs](https://github.com/RNAcentral/R2DT/tree/develop#constraint-based-folding-for-insertions)).

In summary, there are 4 new options for r2dt.py:

```
--constraint
--constraint --fold_type full_molecule
--constraint --fold_type insertions_only
--constraint --fold_type all_constraints_enforced
```

Would it be possible to add support for these flags on wwwdev please? 

I imagine that the UI could have a drop down with 4 elements that correspond to the flags above. Alternatively there could be a checkbox that enables constraints, and if selected, an additional dropdown would enable selection of the `--fold_type`. Ideally the `--constraint` flag alone should work just fine to save the users from selecting the `--fold_type` manually.

@carlosribas @blakesweeney Carlos and Blake, once the API support is implemented, the new options should also be supported in the R2DT UI under _Advanced options_ and show some help text borrowed from the docs linked above.

Please let me know if you have any questions and thank you for working on this!

**Comment by AntonPetrov** [comment]
There is one more new option that I think can be reserved for the command line version of R2DT:

```
--constraint --exclusion <exclusion_file>
```

It requires some advanced validation and error messages which could be a lot of work. If the users ask for it, we can add it in the next iteration. 

@aspetr01 Anton, please shout if you think that this really needs to be integrated in the web application without delay.

**Comment by AntonPetrov** [comment]
Thank you for looking into this Fabio! You are correct, `--fold_type <something>` alone is not functional. 

I think it's okay to have four boolean params if it is easier to implement and maintain, as long as the usage is well documented. 

However, in this case it would be nice to have a check on the API side to make sure that the user did not set more than one of `full_molecule`, `insertions_only`, and `all_constraints_enforced` to `True` (if `--constraint` is specified of course).

**Comment by biomadeira** [comment]
Hi @AntonPetrov,

This is now live in wwwdev. I ended up adding two params as proposed in option two above. The check is simple enough so that looks like a better representation of the CLI's behaviour. From my quick tests, it seems to be working as expected, but have a play with it and let me know!

Best regards

**Comment by AntonPetrov** [comment]
@biomadeira Thank you very much Fabio, this is really cool, I can [see it on wwwdev](https://wwwdev.ebi.ac.uk/Tools/common/tools/help/index.html?tool=r2dt#!/Submit32job/post_run)! 👏

@carlosribas @blakesweeney Carlos and Blake, would you be able to add a "Constrained folding" checkbox to the advanced parameters of the R2DT UI please? Once selected, it should display a dropdown with 3 options for the fold type (see above).

@aspetr01 @hmccann3 Anton and Holly, I remember you had a nice presentation showing different fold type options with inputs and outputs. Would you be able to share it so we can embed it in the R2DT release announcement as a Google Slides deck? Then we could refer the users to the announcement as additional documentation and use the images for a figure in the next paper. What do you think?

Also I've set up the following google doc where we can draft the release announcement for release 1.3, including the constrained folding functionality:
https://docs.google.com/document/d/1cVQWcA0l-iDgDVgtFV29_puYBREgTMpew_ZfzCeIcZo/edit?usp=sharing

Everyone is welcome to contribute to the text! We should be able to reuse it for the next R2DT paper too.

Thank you everybody for working on this! 🚀

**Comment by aspetr01** [comment]
Will do.

**Comment by carlosribas** [comment]
Hi @AntonPetrov

This is now live in [test.rnacentral.org](https://test.rnacentral.org/r2dt). The links will be changed as soon as we update the tool documentation.

It is worth noting that if you click on the advanced option of [this search](https://test.rnacentral.org/r2dt?jobid=r2dt-R20220627-163106-0924-62244155-np2), you will not see the "Constrained folding" option enabled, although it has been used. To show it, we'll need to save this information somewhere, just like we do with the template.

Please give it a try to see if it's working as expected.

Thanks!

**Comment by AntonPetrov** [comment]
Implemented in release 1.3 ✅
